x-pack/plugins/actions/server/create_execute_function.ts

```diff
@@ -113,10 +113,15 @@ export function createBulkExecutionEnqueuerFunction({
       inMemoryConnectors,
       connectorIds
     );
+    const skippedConnectors: string[] = [];
 
-    connectors.forEach((c) => {
+    for (const c of connectors) {
       const { id, connector, isInMemory } = c;
-      validateCanActionBeUsed(connector);
+
+      if (connector.isMissingSecrets) {
+        skippedConnectors.push(id);
+        continue;
+      }
 
       const { actionTypeId } = connector;
       if (!actionTypeRegistry.isActionExecutable(id, actionTypeId, { notifyUsage: true })) {
```

Comment:
the line below this, `actionTypeRegistry.ensureActionTypeEnabled(actionTypeId)`, can throw an error.

I think the `isMissingSecrets` and the actionExecutable tests should go in a single function, that we call from where and try/catch around, to let it throw Errors.

Reply:
done